core/src/test/java/org/jboss/jca/core/connectionmanager/pool/OnePoolNoTxDecrementCapacityMinPoolSizePolicyTestCase.java

```diff
@@ -62,7 +61,18 @@ public void checkPool() throws Exception
    {
       AbstractPool pool = getPool();
       assertEquals(pool.getManagedConnectionPools().size(), 0);
-      fillPoolToSize(5);
+
+      SimpleConnection[] simpleConnections = fillPoolToSize(5);
+
+      // save string representation of ManagedConnection
+      String[] mcHashes = new String[5];
+      for (int i = 0; i < 5; i++)
+      {
+         SimpleConnection sc = simpleConnections[i];
+         SimpleConnectionImpl sci = (SimpleConnectionImpl) sc;
+         mcHashes[i] = sci.getManagedConnection().toString();
```

Comment:
An explicit toString() contract is needed then - e.g. toString() override. Or choose a better contract